packages/react-components/react-combobox/src/utils/useComboboxBaseState.ts

```diff
@@ -71,7 +71,7 @@ export const useComboboxBaseState = (
     // we do not want to accidentally override defaultValue on a second render
     // unless another value is intentionally set
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [controllableValue, editable, getOptionsMatchingValue, multiselect, props.defaultValue, selectedOptions]);
+  }, [controllableValue, editable, getOptionsMatchingValue, multiselect, selectedOptions]);
```

Comment:
This is a fragile way to implement this behavior. If another dependency changes, then it'll recalculate the value using a potentially new `defaultValue`, leading to confusing (and hard to debug) behavior.

Additionally, React does not guarantee that the value from `useMemo` will be saved in all circumstances. From https://react.dev/reference/react/useMemo:
> ### 📄 Note
> **You should only rely on `useMemo` as a performance optimization.** If your code doesn’t work without it, find the underlying problem and fix it first. Then you may add `useMemo` to improve performance.

There are possible solutions, like saving the original `defaultValue` in a ref, and relying only on its original value. However, I'm not sure if that's the right solution here.